src/v/datalake/translation/partition_translator.cc

```diff
@@ -184,13 +184,20 @@ ss::future<> partition_translator::translate_until_stopped() {
         // is NOT synchronized with outstanding commit operations. Therefore
         // if we reach this point before the most recent batch of files has
         // been committed, the commit lag metric will be out of sync at least
-        // until `wait_for_data_to_translate` wakes up.
+        // until 'wait_for_data' returns and we re-enter the loop.
         _data_source->update_commit_lag(last_committed_offset);
         _data_source->update_translation_lag(last_translated_offset);
 
         // Wait until some data is ready to be translated.
-        auto begin_offset = co_await _data_source->wait_for_data_to_translate(
-          last_translated_offset, _as);
+        auto maybe_begin_offset
+          = co_await _data_source->wait_for_data_to_translate(
+            last_translated_offset, _as);
+
+        if (!maybe_begin_offset.has_value()) {
+            continue;
```

Comment:
nit: I think a comment would be helpful here to mention the motivation behind continuing (like as is, it's totally unclear that this is important for metrics, but i also imagine this could be useful if there's some mismatch between the coordinator and the translator?)